src/Microsoft.Azure.SignalRService/ServiceManagerOptionsSetup.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalRService
+{
+    internal class ServiceManagerOptionsSetup : IConfigureOptions<ServiceManagerOptions>
+    {
+        private IConfigurationRoot _configuration;
+
+        public void Configure(ServiceManagerOptions options)
+        {
+            _configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appSettings.json", optional: true, reloadOnChange: true)
+                .AddEnvironmentVariables()
+                .AddUserSecrets<ServiceManagerBuilder>(optional: true)
+                .Build();
+
+            options.ConnectionString = _configuration[ServiceManagerOptions.ConnectionStringDefaultKey];
+
+            if (int.TryParse(_configuration[ServiceManagerOptions.ConnectionCountDefaultKey], out int connectionCount))
+            {
+                options.ConnectionCount = connectionCount;
+            }
```

Comment:
Do you also want to set transport type here?

Reply:
class removed